### PR TITLE
Add codespell to the pre commit config #2109 

### PR DESCRIPTION
**Pull request by anufdo**
Add codespell to the pre commit

### CODE REVIEW

File: .codespellrc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[codespell]
+ignore-words-list:
+  - codespell.txt
+-I FILE:
+  - docs/*
```
Comment (jbampton) [inline .codespellrc:5]:
```suggestion
  - doc/*
```

**Review by jbampton — CHANGES_REQUESTED** [review]
Will need to fix the spelling in the `doc` folder not `docs`.

See https://github.com/slurpcode/slurp/pull/2111/files

File: .codespellrc (hunk 1 above)
Comment (anufdo) [reply .codespellrc:5]: Yes.

**Comment by jbampton** [comment]
https://github.com/One-Language/One/blob/main/CONTRIBUTING.md#pre-commit

File: .codespellrc (hunk 1 above)
Comment (anufdo) [reply .codespellrc]: https://github.com/slurpcode/slurp/pull/2112/commits/8ad8ad05bdef64214b141e87142b3752ba598d24

**Comment by jbampton** [comment]
https://github.com/slurpcode/slurp/actions/runs/4533746250/jobs/7986975862?pr=2112

**Comment by anufdo** [comment]
Now it is scanning doc folder.
Is it now Ok to set codespell to fix spilling errors automatically?
It will change lot of files automatically.

**Comment by jbampton** [comment]
You can use:

`slurp % codespell doc/source | cut -f2 -d' ' | tr A-Z a-z | sort | uniq`

as a starting point for the ignore list.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ default_stages: [commit, push]
 default_language_version:
   # force all unspecified Python hooks to run python3
   python: python3
-minimum_pre_commit_version: '1.20.0'
+minimum_pre_commit_version: "1.20.0"
```
Comment (jbampton) [inline .pre-commit-config.yaml:6]:
```suggestion
minimum_pre_commit_version: '1.20.0'
```

**Review by jbampton — APPROVED** [review]

**Pull request merged**